code/API_definitions/Application_Endpoint_Discovery_API.yaml

```diff
@@ -0,0 +1,596 @@
+    ```
+    # Authorization and authentication
+    In order to maintain the integrity and security of the Application Discovery
+    API, access is governed by robust authentication and authorization
+    processes.
+    This ensures that only valid, authorized requests are able to interact with
+    the API and retrieve application endpoint information.
+    ## OpenID Connect
+    The API utilizes OpenID Connect on top of OAuth 2.0 for user authentication.
+    Users must register to obtain client credentials, which are necessary to
+    acquire access tokens from the OpenID provider.
+    ## Obtaining Access Tokens
+    Access tokens are obtained by submitting a request to the designated OpenID
+    Connect URL, utilizing the client credentials. It is imperative that the
+    retrieved token is included in the `Authorization` header as a `Bearer`
+    token for all subsequent API requests.
+    ## Scopes and Permissions
+    During the token request, developers will need to specify the scopes that
+    correspond to the permissions the application requires. Our API strictly
+    enforces scope validation to ensure clients only perform operations that
+    are within their granted permissions.
+    ## Secure Transmission
+    All communication with the API, including token requests, must occur over
+    HTTPS to ensure data confidentiality and protection.
+    ## Error Handling
+    If a request is made without a token or with an expired or invalid token,
+    the API will return a  `401 Unauthorized` status code. Similarly, if the
+    token does not include the proper scopes for the requested operation,
+    a `403 Forbidden` status code will be returned, indicating
+    insufficient permissions.
+    ---
+  contact:
+    email: sp-edc@lists.camaraproject.org
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+externalDocs:
+  description: Product documentation at CAMARA.
+  url: https://github.com/camaraproject/EdgeCloud
+servers:
+  - url: "{apiRoot}/{basePath}"
+    variables:
+      apiRoot:
+        default: https://localhost:443
+        description: API root.
+      basePath:
+        default: application-endpoint-discovery/vwip
+        description: Base path for the Application Endpoint Discovery.
+tags:
+  - name: Discovery
+    description: |
+      Find the closest App Instance Endpoint to the user device.
+paths:
+  /apps-endpoints:
+    get:
+      security:
+        - openId:
+            - application-endpoint-discovery:apps-endpoints:read
+      operationId: getAppEndpoints
+      parameters:
+        - name: filter
+          in: query
+          required: true
+          description: |
+            Filter the Edge Cloud Zones according to the parameter value.
+            For this API the only supported value is `closest`.
+          schema:
+            type: string
+            enum:
+              - closest
+        - name: IPv4-Address
+          in: header
```

Comment:
I think we should mention about what are valid combination from the give set of the user identifiers e.g. IPv4-Address, Network-Access-Identifier etc. If one of them or any of them or all should be present?

Reply:
OK. Agree with you. We are going to wait for the new commonalities to come out for guidance.